libraries/chain/include/scorum/chain/data_service_factory.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <fc/time.hpp>
+#include <fc/optional.hpp>
+#include <scorum/protocol/types.hpp>
+
+namespace scorum {
+namespace chain {
+
+class database;
+class dbservice_dbs_factory;
+
+class account_service_i;
+class proposal_service_i;
+class committee_service_i;
+class property_service_i;
+
+class data_service_factory_i
```

Comment:
'struct' would be better for interface. All methods are public.

Reply:
I don't like mixing classes and structs. I would prefer  to leave a class.